app/models/designation.js

```diff
@@ -61,6 +62,58 @@ export default Model.extend({
     return this.get("state").capitalize();
   }),
 
+  stateIcon: Ember.computed('stateIcon', function () {
+    const state = this.get("state");
+    switch (state) {
+      case "awaiting_dispatch":
+      case "scheduled":
+        return "clock-o";
+      case "processing":
+        return "list";
+      case "submitted":
+        return "envelope";
+      case "dispatching":
+        return "paper-plane";
+      case "cancelled":
+        return "thumbs-down";
+      case "closed":
+        return "lock";
+      default:
+        return "";
+    }
+  }),
+
+  transportIcon: Ember.computed("transportIcon", function() {
```

Comment:
@patrixr , I think this computed property should be dependent upon `transportKey `